tgui/packages/tgui/interfaces/Uplink.js

```diff
@@ -0,0 +1,188 @@
+        </Tabs>
+        {PickTab(tabIndex)}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const ItemsPage = (_properties, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    crystals,
+    cats,
+  } = data;
+  // Default to first
+  const [
+    uplinkCat,
+    setUplinkCat,
+  ] = useLocalState(context, 'uplinkTab', cats[0]);
+  return (
+    <Section
+      title={"Current Balance: " + crystals + "TC"}
+      buttons={
+        <Fragment>
+          <Button
+            content="Random Item"
+            icon="question"
+            onClick={() => act('buyRandom')}
+          />
+          <Button
+            content="Refund Currently Held Item"
+            icon="undo"
+            onClick={() => act('refund')}
+          />
+        </Fragment>
+      }>
+      <Flex>
+        <FlexItem>
+          <Tabs vertical>
+            {cats.map(c => (
+              <Tabs.Tab
+                key={c}
+                selected={c === uplinkCat}
+                onClick={() => setUplinkCat(c)}>
+                {c.cat}
+              </Tabs.Tab>
+            ))}
+          </Tabs>
+        </FlexItem>
+        <Flex.Item grow={1} basis={0}>
+          {uplinkCat.items.map(i => (
+            <Section
+              key={decodeHtmlEntities(i.name)}
+              title={decodeHtmlEntities(i.name)}
+              buttons={
+                <Button
+                  content={"Buy (" + i.cost + "TC)" + (i.refundable ? " [Refundable]" : "")}
+                  color={i.hijack_only === 1 && "red"}
+                  // Yes I care this much about both of these being able to render at the same time
+                  tooltip={(i.hijack_only === 1 && "Hijack Agents Only!")}
+                  tooltipPosition="left"
+                  onClick={() => act("buyItem", {
+                    item: i.obj_path,
+                  })}
+                  disabled={i.cost > crystals}
+                />
+              }>
+              <Box italic>
+                {decodeHtmlEntities(i.desc)}
+              </Box>
+            </Section>
+          ))}
+        </Flex.Item>
+      </Flex>
+    </Section>
+  );
+};
+
+
+const ExploitableInfoPage = (_properties, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    exploitable,
+  } = data;
+  // Default to first
+  const [
+    selectedRecord,
+    setSelectedRecord,
+  ] = useLocalState(context, 'selectedRecord', exploitable[0]);
+
+  const [
+    searchText,
+    setSearchText,
+  ] = useLocalState(context, 'searchText', '');
+
+  // Search for peeps
+  const SelectMembers = (people, searchText = '') => {
+    const MemberSearch = createSearch(searchText, member => member.name);
+    return flow([
+      // Null camera filter
+      filter(member => member?.name),
+      // Optional search term
+      searchText && filter(MemberSearch),
+      // Slightly expensive, but way better than sorting in BYOND
+      sortBy(member => member.name),
+    ])(people);
+  };
```

Comment:
Since it's the second time I'm seeing this bit of code - it might be worthwhile to make it common as to avoid duplicate code as much as possible.

Reply:
The params change slightly with each, so its staying like this